crates/turbo-tasks-memory/src/task.rs

```diff
@@ -508,7 +463,49 @@ impl Task {
                 )
             }
         };
-        true
+        let future = match &self.ty {
```

Comment:
This makes the `execution_started` even larger and harder to parse. Move this to a `TaskType::execute` method instead. It no longer uses `self`, everything is in `TaskType`.

Reply:
It uses `state` from `self`, but I can move it into some different method in Task